### PR TITLE
enh(platformstatus): add hostname to platform topology

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

Add real server hostname to platform_topology database
**Fixes**  MON-6071

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x (master)

## Checklist

- [ ] I followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/install/sql/centreon/Update-DB-20.10.0-beta.2.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+ALTER TABLE `platform_topology` ADD COLUMN `hostname` varchar(255) NULL AFTER `address`;
```
Comment (sc979) [inline www/install/sql/centreon/Update-DB-20.10.0-beta.2.sql:1]: kindly, if you can resolve the hostname and add them for remote or central in the upgrade script

**Review by kduret — DISMISSED** [review]

File: www/install/php/Update-20.10.0-beta.2.post.php [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+//error specific content
+$versionOfTheUpgrade = 'UPGRADE - 20.10.0-beta.2.post : ';
+
+/**
+ * Queries needing exception management and rollback if failing
+ */
+try {
+    $pearDB->beginTransaction();
+    $pearDB->query("TRUNCATE TABLE `platform_topology`");
```
Comment (sc979) [inline www/install/php/Update-20.10.0-beta.2.post.php:33]: JAC : truncate does not support rollback. You can move it to the sql file

File: www/install/php/Update-20.10.0-beta.2.post.php [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    // Check if the server is enabled
+    $errorMessage = "Unable to find the server in 'nagios_server' table.";
+    $serverQuery = $pearDB->query("
+        SELECT `id`, `name` FROM nagios_server
+        WHERE localhost = '1' AND ns_activate = '1'
+    ");
+
+    $bindValues = [];
+    $hostName = gethostname();
+    $hostName
```
Comment (sc979) [inline www/install/php/Update-20.10.0-beta.2.post.php:56]:
```suggestion
    !empty($hostName)
```

Comment (jeremyjaouen) [reply www/install/php/Update-20.10.0-beta.2.post.php:56]: gethostname() will return the hostname or FALSE

Comment (sc979) [reply www/install/php/Update-20.10.0-beta.2.post.php:56]: If the hostname is empty, Will you insert null or ''

File: www/install/php/Update-20.10.0-beta.2.post.php [hunk 4]
```diff
@@ -30,7 +30,7 @@
  */
 try {
     $pearDB->beginTransaction();
-    $pearDB->query("TRUNCATE TABLE `platform_topology`");
+    $pearDB->query("DELETE FROM `platform_topology`");
```
Comment (sc979) [inline www/install/php/Update-20.10.0-beta.2.post.php:33]:
```suggestion
    // Remove data inserted in 20.10.0-beta1
    $pearDB->query("DELETE FROM `platform_topology`");
```

File: www/install/steps/process/insertBaseConf.php [hunk 5]
```diff
@@ -91,25 +91,37 @@
...
             NULL,
             :id
         )
     ");
     $stmt->bindValue(':centralAddress', $_SERVER['SERVER_ADDR'], \PDO::PARAM_STR);
+    foreach ($bindValues as $token => $bindParams) {
+        foreach ($bindParams as $paramType => $paramValue) {
+            $stmt->bindValue($token, $paramValue, $paramType);
+        }
+    }
```
Comment (sc979) [inline www/install/steps/process/insertBaseConf.php:124]:
```suggestion
    $stmt->bindValue(':centralAddress', $_SERVER['SERVER_ADDR'], \PDO::PARAM_STR);
    $hostName
        ? $stmt->bindValue(':hostname', $hostName, \PDO::PARAM_STR)
        : $stmt->bindValue(':hostname', null, \PDO::PARAM_NULL);
```

Comment (sc979) [reply www/install/steps/process/insertBaseConf.php:124]: no need for a foreach on foreach loop for one value

File: www/install/steps/process/insertBaseConf.php [hunk 6]
```diff
@@ -91,25 +91,37 @@
     exit;
 }
 
+$bindValues = [];
+$hostName = gethostname();
+$hostName
+    ? $bindValues[':hostname'] = [\PDO::PARAM_STR => $hostName]
+    : $bindValues[':hostname'] = [\PDO::PARAM_NULL => null];
```
Comment (sc979) [inline www/install/steps/process/insertBaseConf.php:98]:
```suggestion
```

File: www/install/steps/process/insertBaseConf.php [hunk 7]
```diff
@@ -91,25 +91,32 @@
     exit;
 }
 
+$bindValues = [];
```
Comment (sc979) [inline www/install/steps/process/insertBaseConf.php:94]:
```suggestion
```

**Review by sc979 — DISMISSED** [review]

File: www/install/steps/process/insertBaseConf.php [hunk 8]
```diff
@@ -91,25 +91,31 @@
     exit;
 }
 
+$hostName = gethostname();
```
Comment (kduret) [inline www/install/steps/process/insertBaseConf.php:94]:
```suggestion
$hostName = gethostname() ?: null;
```

File: www/install/steps/process/insertBaseConf.php [hunk 9]
```diff
@@ -91,25 +91,31 @@
...
             :name,
             'central',
             NULL,
             :id
         )
     ");
     $stmt->bindValue(':centralAddress', $_SERVER['SERVER_ADDR'], \PDO::PARAM_STR);
+    $hostName
+        ? $stmt->bindValue(':hostname', $hostName, \PDO::PARAM_STR)
+        : $stmt->bindValue(':hostname', null, \PDO::PARAM_NULL);
```
Comment (kduret) [inline www/install/steps/process/insertBaseConf.php:118]:
```suggestion
    $stmt->bindValue(':hostname', $hostName, \PDO::PARAM_STR);
```

File: www/install/php/Update-20.10.0-beta.2.post.php [hunk 10]
```diff
@@ -0,0 +1,134 @@
...
+    $serverQuery = $pearDB->query("
+        SELECT `id`, `name` FROM nagios_server
+        WHERE localhost = '1' AND ns_activate = '1'
+    ");
+
+    $bindValues = [];
+    $hostName = gethostname();
+    $hostName
+        ? $bindValues[':hostname'] = [\PDO::PARAM_STR => $hostName]
+        : $bindValues[':hostname'] = [\PDO::PARAM_NULL => null];
```
Comment (kduret) [inline www/install/php/Update-20.10.0-beta.2.post.php:59]:
```suggestion
    $hostName = gethostname() ?: null;
    $bindValues[':hostname'] = [\PDO::PARAM_STR => $hostName];
```

Comment (sc979) [reply www/install/php/Update-20.10.0-beta.2.post.php:59]: @jeremyjaouen ... it seems it's the right time to remove the foreach, foreach loop ;)

File: www/install/php/Update-20.10.0-beta.2.post.php (hunk 3 above)
Comment (callapa) [reply www/install/php/Update-20.10.0-beta.2.post.php:57]: The use of a ternary operator is not appropriate here.

File: www/install/php/Update-20.10.0-beta.2.post.php [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+        }
+        $stmt->bindValue(':type', $type, \PDO::PARAM_STR);
+        $stmt->bindValue(':id', (int)$row['id'], \PDO::PARAM_INT);
+        $stmt->execute();
+    }
+
+    // get topology local server id
+    $localStmt = $pearDB->query(
+        "SELECT `id` FROM `platform_topology`
+        WHERE `server_id` = (SELECT `id` FROM nagios_server WHERE localhost = '1')"
```
Comment (callapa) [inline www/install/php/Update-20.10.0-beta.2.post.php:83]: Why don't you use a JOIN?

File: www/install/php/Update-20.10.0-beta.2.post.php [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+    while ($row = $childStmt->fetch()) {
+        //check remote/poller
+        $parent = $parentId;
+        $serverType = 'poller';
+        if ($row['remote_id']) {
+            $parent = $row['remote_id'];
+        }
+
+        $remoteServerQuery = $pearDB->query(
+            "SELECT `id` FROM remote_servers WHERE ip = '" . $row['ns_ip_address'] . "'"
```
Comment (callapa) [inline www/install/php/Update-20.10.0-beta.2.post.php:101]: Prepare your SQL request please.

File: www/install/php/Update-20.10.0-beta.2.post.php [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+include_once __DIR__ . "/../../class/centreonLog.class.php";
+$centreonLog = new CentreonLog();
```
Comment (callapa) [inline www/install/php/Update-20.10.0-beta.2.post.php:23]: Can be moved directly in the catch expression.

**Review by callapa — CHANGES_REQUESTED** [review]

**Review by sc979 — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**